app/client/apps/goat/lib/store/styling/slice.ts

```diff
@@ -19,22 +20,56 @@ interface IBasemap {
   thumbnail: string;
 }
 
+interface IMarker {
+  id: string;
+  lat: number;
+  long: number;
+  iconName: string;
+}
+
+interface IMapLayer {
+  name: string;
+  sources: {
+    composite: {
+      url: string;
+      type: string;
+    };
+  };
+  id: string;
+  type:
+    | "symbol"
+    | "fill"
+    | "line"
+    | "circle"
+    | "background"
+    | "raster"
+    | "fill-extrusion"
+    | "heatmap"
+    | "hillshade";
+  paint: object;
+  layout: object;
+  source: string;
+  "source-layer": string;
+}
+
```

Comment:
This type should be part of the react-map-gl, so we don't need to create it again. 
https://visgl.github.io/react-map-gl/docs/api-reference/layer
https://github.com/visgl/react-map-gl/blob/7.1-release/src/types/index.ts

Reply:
@majkshkurti This is a mocked layer type, I will fix it as you mentioned when will get the layer from db. At that time some changes will be required in code

Reply:
I see. Can't we use the react-map-gl type in this case ? I mean it's the same type, right ?

Reply:
@majkshkurti  I don't know the correct structure for making mock data.
This is my mocked data for the fill layer. But in the FillLayer type, there is a difference. To use  FillLayer I need to fix many types of issues but I don't know the correct structure of the layer.
{
    name: "aoi",
    sources: {
      composite: {
        url: "mapbox://eliaspajares.cljxc2rek01ow2alyl0cy0y2j-63c9z",
        type: "vector",
      },
    },
    id: "cljyel7yl005r01pfcd4h0epj",
    type: "fill",
    paint: {
      "fill-color": "#316940",
      "fill-outline-color": "#000",
      "fill-opacity": 1.0,
      "fill-antialias": true, //
    },
    layout: {},
    source: "composite",
    "source-layer": "aoi",
  }